Content.Server/GameObjects/Components/GUI/StrippableComponent.cs

```diff
@@ -0,0 +1,371 @@
+                NeedHand = true,
+            };
+
+            var result = await doAfterSystem.DoAfter(doAfterArgs);
+            if (result != DoAfterStatus.Finished) return;
+
+            userHands.Drop(hand, false);
+            hands.PutInHand(item, hand, false);
+            UpdateSubscribed();
+        }
+
+        /// <summary>
+        ///     Takes an item from the inventory and places it in the user's active hand.
+        /// </summary>
+        private async void TakeItemFromInventory(IEntity user, EquipmentSlotDefines.Slots slot)
+        {
+            var inventory = Owner.GetComponent<InventoryComponent>();
+            var userHands = user.GetComponent<HandsComponent>();
+
+            bool Check()
+            {
+                if (!ActionBlockerSystem.CanInteract(user))
+                    return false;
+
+                if (!inventory.HasSlot(slot))
+                    return false;
+
+                if (!inventory.TryGetSlotItem(slot, out ItemComponent itemToTake))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("{0:They} {0:have} nothing there!", Owner));
+                    return false;
+                }
+
+                if (!inventory.CanUnequip(slot))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("{0:They} cannot unequip that!", Owner));
+                    return false;
+                }
+
+                return true;
+            }
+
+            var doAfterSystem = EntitySystem.Get<DoAfterSystem>();
+
+            var doAfterArgs = new DoAfterEventArgs(user, StripDelay, CancellationToken.None, Owner)
+            {
+                ExtraCheck = Check,
+                BreakOnStun = true,
+                BreakOnDamage = true,
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+            };
+
+            var result = await doAfterSystem.DoAfter(doAfterArgs);
+            if (result != DoAfterStatus.Finished) return;
+
+            var item = inventory.GetSlotItem(slot);
+            inventory.Unequip(slot);
+            userHands.PutInHandOrDrop(item);
+            UpdateSubscribed();
+        }
+
+        /// <summary>
+        ///     Takes an item from a hand and places it in the user's active hand.
+        /// </summary>
+        private async void TakeItemFromHands(IEntity user, string hand)
+        {
+            var hands = Owner.GetComponent<HandsComponent>();
+            var userHands = user.GetComponent<HandsComponent>();
+
+            bool Check()
+            {
+                if (!ActionBlockerSystem.CanInteract(user))
+                    return false;
+
+                if (!hands.HasHand(hand))
+                    return false;
+
+                if (!hands.TryGetItem(hand, out var heldItem))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("{0:They} {0:have} nothing there!", Owner));
```

Comment:
This popup incorrectly displays `"He Urist Mchands (3569, HumanMob_Content nothing there!"`.